File: java/AdropAdsExample/app/src/main/java/io/adrop/ads/example/helper/ErrorUtils.java
package io.adrop.ads.example.helper;

import io.adrop.ads.model.AdropErrorCode;

public class ErrorUtils {
    public static String descriptionOf(AdropErrorCode errorCode) {
        switch (errorCode) {
            case ERROR_CODE_NETWORK:
                return "The network status is unstable";
            case ERROR_CODE_INTERNAL:
                return "Exception in SDK";
            case ERROR_CODE_INITIALIZE:
                return "Initialize Adrop first before you act something";
            case ERROR_CODE_INVALID_UNIT:
                return "Ad unit ID is not valid";
            case ERROR_CODE_AD_INACTIVE:
                return "There are no active advertising campaigns";
            case ERROR_CODE_AD_NO_FILL:
                return "Unable to receive ads that meet the criteria. Please retry";
            case ERROR_CODE_AD_LOAD_DUPLICATED:
                return "You can't load again after ad received successfully";
            case ERROR_CODE_AD_LOADING:
                return "Waiting ad response from server after request ad";
            case ERROR_CODE_AD_EMPTY:
                return "There is no ad received";
            case ERROR_CODE_AD_SHOWN:
                return "This ad was shown already";
            case ERROR_CODE_AD_HIDE_FOR_TODAY:
                return "You can't load ad for today.";
            case ERROR_CODE_ACCOUNT_USAGE_LIMIT_EXCEEDED:
                return "App key (account) usage exceeded limit";
            default:
                return "";
        }
    }
}

File: kotlin/AdropAdsExample/app/src/main/java/io/adrop/ads/example/helper/ErrorUtils.kt
package io.adrop.ads.example.helper

import io.adrop.ads.model.AdropErrorCode

object ErrorUtils {

    @JvmStatic
    fun descriptionOf(errorCode: AdropErrorCode?): String {
        return when (errorCode) {
            AdropErrorCode.ERROR_CODE_NETWORK -> "The network status is unstable"
            AdropErrorCode.ERROR_CODE_INTERNAL -> "Exception in SDK"
            AdropErrorCode.ERROR_CODE_INITIALIZE -> "Initialize Adrop first before you act something"
            AdropErrorCode.ERROR_CODE_INVALID_UNIT -> "Ad unit ID is not valid"
            AdropErrorCode.ERROR_CODE_AD_INACTIVE -> "There are no active advertising campaigns"
            AdropErrorCode.ERROR_CODE_AD_NO_FILL -> "Unable to receive ads that meet the criteria. Please retry"
            AdropErrorCode.ERROR_CODE_AD_LOAD_DUPLICATED -> "You can't load again after ad received successfully"
            AdropErrorCode.ERROR_CODE_AD_LOADING -> "Waiting ad response from server after request ad"
            AdropErrorCode.ERROR_CODE_AD_EMPTY -> "There is no ad received"
            AdropErrorCode.ERROR_CODE_AD_SHOWN -> "This ad was shown already"
            else -> ""
        }
    }
}
